migrate/20240116_move_fw_rules_to_vm.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    create_table(:firewall) do
+      column :id, :uuid, primary_key: true
+      column :name, :text, null: false, default: "Default"
+      column :description, :text, null: false, default: "Default firewall"
+      column :created_at, :timestamp, null: false, default: Sequel::CURRENT_TIMESTAMP
+      foreign_key :vm_id, :vm, type: :uuid, null: true
+    end
+
+    alter_table(:firewall_rule) do
+      add_foreign_key :firewall_id, :firewall, null: true, type: :uuid
+      set_column_allow_null :private_subnet_id
+      add_unique_constraint [:ip, :port_range, :firewall_id]
+    end
+
+    run <<~SQL
+INSERT INTO firewall (id, name, description, vm_id)
+SELECT id, 'Default', 'Default firewall', id FROM vm;
+    SQL
+
+    # populate firewall rules per vm
+    run <<~SQL
+WITH vm_subnet_mapping AS (
+  SELECT
+      vm.id AS vm_id,
+      nic.private_subnet_id
+  FROM
+      vm
+  INNER JOIN
+      nic ON vm.id = nic.vm_id
+),
+new_firewall_rules AS (
+    SELECT
+        gen_random_uuid() AS id, -- Generate new UUIDs for the id
+        fwr.ip as ip,
+        fwr.port_range as port_range,
+        vm_subnet_mapping.vm_id AS firewall_id
+    FROM
+        firewall_rule fwr
+    INNER JOIN
+        vm_subnet_mapping ON vm_subnet_mapping.private_subnet_id = fwr.private_subnet_id
+)
+INSERT INTO firewall_rule (id, ip, port_range, firewall_id)
+SELECT id, ip, port_range, firewall_id FROM new_firewall_rules;
+    SQL
+
+    run <<~SQL
+DELETE FROM firewall_rule
+WHERE private_subnet_id IS NOT NULL;
+    SQL
+
+    alter_table(:firewall_rule) do
+      drop_column :private_subnet_id
+      set_column_not_null :firewall_id
+    end
```

Comment:
I am planning to move these into a second commit that is at the very end of this PR, so that the migration is graceful